go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -286,7 +290,6 @@ func runOvnKube(ctx *cli.Context) error {
 	// start the prometheus server to serve OVN Metrics (default port: 9476)
 	// Note: for ovnkube node mode dpu-host no ovn metrics is required as ovn is not running on the node.
 	if config.OvnKubeNode.Mode != types.NodeModeDPUHost && config.Kubernetes.OVNMetricsBindAddress != "" {
-		metrics.RegisterOvnMetrics(ovnClientset.KubeClient, node)
```

Comment:
@girishmg iirc you put the metrics in the node because in upstream/nvidia you run OVN DB on its own node so running it on master isnt feasible. Is my memory correct? Maybe we should put the ovn metrics into ovn-dbchecker?

Reply:
@trozet that is correct. To enable rolling updates, this is the split we have:


```
|-------------+-----------------------------+-----------|
| Kind        | Pod                         | Component |
|-------------+-----------------------------+-----------|
| Statefulset | ovn-nbdb                    | OVN       |
| Statefulset | ovn-sbdb                    | OVN       |
| Daemonset   | ovn-host                    | OVN       |
| Deployment  | ovn-northd (replicas=3)     |           |
|-------------+-----------------------------+-----------|
| Deployment  | ovnkube-master (replicas=3) | OVN-K8s   |
| Daemonset   | ovnkube-node                | OVN-K8s   |
|-------------+-----------------------------+-----------|
```

There was also requirement from both RH and NVDA that we reduce the number of TCP Ports from which we expose prometheus metrics. If we have more TCP Ports, then we need equal instances of rbac-proxy to terminate TLS. So, we moved all the OVN/OVS/Ovnkube-node metric collection to ovnkube-node.

Ovnkube-master still continues to export it's metric

Reply:
@girishmg what do you think about putting the metrics on ovndbchecker?